merge.py

```diff
@@ -1,98 +1,196 @@
-  if not mvn_success == 0:
-    post_error('mvn clean verify against HEAD + PR#{} failed. Not merging.'.format(pr), pr)
-    return False
-
-  # git push (COMMENTED UNTIL MERGEBOT HAS PERMISSIONS)
-  #push_success = call(['git', 'push', 'apache', 'HEAD:master'], cwd='/usr/local/google/home/jasonkuster/tmp/')
-  #if not push_success == 0:
-  #  post_error('Git push failed. Please try again.', pr)
-  #  return False
-  return True
-
-def set_up():
-  if not call(['mkdir', '/usr/local/google/home/jasonkuster/tmp']) == 0:
-    return False
-  return True
-
-
-def clean_up():
-  if not call(['rm', '-rf', '/usr/local/google/home/jasonkuster/tmp']) == 0:
-    return False
-  return True
 
+def create_merger(config, work_queue):
+    """create_merger creates a merger of the type specified in the config.
+
+    Args:
+        config: A dictionary containing repository configuration.
+        work_queue: A Queue.Queue where new work items will be queued.
+    Returns:
+        Merger of type specified in configuration
+    Raises:
+        AttributeError: if passed an unsupported SCM type.
+    """
+    if config['scm_type'] == 'github':
+        return GitMerger(config, work_queue)
+    raise AttributeError('Unsupported SCM type: {}.'.format(config['scm_type']))
+
+
+class Merger(object):
+    """Merger is the base class for all mergers.
+    """
+
+    def __init__(self, config, work_queue):
+        self.config = config
+        if not isinstance(work_queue, Queue.Queue):
+            raise AttributeError('Expected work_queue to be of type '
+                                 'Queue.Queue.')
+        self.work_queue = work_queue
+
+    def merge(self):
+        """merge spins forever, merging things off of the work queue.
+        """
+        raise NotImplementedError('merge is required to be implemented.')
+
+
+class GitMerger(Merger):
+    """GitMerger merges Git pull requests.
+    """
+
+    def merge(self):
+        """merge spins forever, merging things off of the work queue.
+
+        Raises:
+            AttributeError: if work items are not of type github_helper.GithubPR
+        """
+        while True:
+            pr = self.work_queue.get()
+            if not isinstance(pr, github_helper.GithubPR.__class__):
+                raise AttributeError('Expected items in work_queue to be of '
+                                     'type github_helper.GithubPR')
+            self.merge_git_pr(pr)
+
+    def merge_git_pr(self, pr):
+        """merge_git_pr merges git pull requests.
+
+        Args:
+            pr: The GithubPR to merge.
+        Returns:
+            True if merging was successful.
+            False otherwise.
+        """
+        repo = self.config['repository']
+        tmp_dir = TMP_DIR_FMT.format('{}-{}'.format(repo, pr.get_num()))
+        if not set_up(tmp_dir):
+            pr.post_error('Error setting up - please try again.')
+            return False
+
+        # Clone repository and configure.
```

Comment:
From a testing/readibility perspective consider creating a bunch of helper methods for each of the big steps, rather than just comments and separate blocks. It will allow testing each piece separately, and also serve to document the steps just as well as the separate blocks. Also enables future refactoring moving some of those methods to other classes.